Bottom: 01dd98b271d595b0c365c50f065935ca02100d3f
Top:    431fe76cbf6de6ef7093ecc7973016feca6529ac
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-03-15 10:30:54 +0800

Refresh of virtio-net

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 14a65eb..3ce5f37 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -810,9 +810,8 @@ bool virtnet_frag_refill(struct receive_queue *rq, unsigned int sz,
 		put_page(pfrag->page);
 	}
 
-	/* FIXME: check GFP_ATOMIC and bypass the cache */
 	pfrag->offset = 0;
-	if (rq->page_cache.index > 0) {
+	if (gfp == GFP_ATOMIC && rq->page_cache.index > 0) {
 		pfrag->page = rq->page_cache.pages[--rq->page_cache.index];
 		pfrag->size = PAGE_SIZE << compound_order(pfrag->page);
 		return true;
